dashboard/app/views/layouts/_small_footer.html.haml

```diff
@@ -1,24 +1,32 @@
-.small-footer
-  - if view_options[:has_i18n]
-    = form_tag(locale_url, method: :post, id: 'localeForm', style: 'margin-bottom: 0px;') do
-      = hidden_field_tag :return_to, request.url
-      = select_tag :locale, options_for_select(options_for_locale_select, locale), onchange: 'this.form.submit();'
-  %small
-    = link_to I18n.t('footer.copyright'), '#', class: 'copyright-link'
-    &nbsp;&nbsp;|&nbsp;&nbsp;
-    %a{:href => "#", :class => "more-link"}
-      = I18n.t('footer.more')
-      %i{:class => "fa fa-caret-up"}
+= javascript_include_tag 'react'
 
-#copyright-flyout
-  %p!= I18n.t('footer.thank_you')
-  %p= I18n.t('footer.help_from_html')
-  %p!= I18n.t('footer.art_from_html', current_year: Time.now.year)
-  %p= I18n.t('footer.powered_by_aws')
-  %p!= I18n.t('footer.trademark', current_year: Time.now.year)
+#page-small-footer
 
-%ul#more-menu
-  %li= link_to I18n.t('footer.support'), 'https://support.code.org'
-  %li= link_to I18n.t('footer.translate'), 'https://code.org/translate'
-  %li= link_to I18n.t('footer.tos'), 'https://code.org/tos'
-  %li= link_to I18n.t('footer.privacy'), 'https://code.org/privacy'
+:javascript
```

Comment:
OK, good point -- since all you're doing here is passing parameters to a react component, I could believe that it's ok to use inline JS in this case. Furthermore, it is hard to think of any error you could make in this JS that wouldn't be caught by the PropTypes check in the React component. Perhaps we should be lenient in this way only when instantiating React components, to encourage people to use React :-)